### PR TITLE
Hash: Drop support for deserializing the legacy format

### PR DESCRIPTION
**Pull request by fviernau**
The model for the `Hash` class has been changed in 2019.
Use the new format throughout the code base and drop handling of the legacy format to reduce technical debt.

### CODE REVIEW

File: integrations/schemas/curations-schema.json [hunk 1]
```diff
@@ -85,30 +65,10 @@
...
-              ]
-            },
-            "else": {
-              "required": [
-                "hash",
-                "hash_algorithm"
-              ]
-            },
             "required": [
+              "hash",
```
Comment (sschuberth) [inline integrations/schemas/curations-schema.json:71]: I believe this commit should also drop the `"string"` type from https://github.com/oss-review-toolkit/ort/blob/ca7beff23342d5c1b9a2bdbaa6c79c9e6cadadeb/integrations/schemas/curations-schema.json#L133.

**Review by sschuberth — CHANGES_REQUESTED** [review]

**Review by sschuberth — DISMISSED** [review]

**Comment by fviernau** [comment]
@oss-review-toolkit/bosch, @bennati is this ok to merge from Bosch and HERE side?

**Comment by nnobelis** [comment]
> @oss-review-toolkit/bosch, @bennati is this ok to merge from Bosch and HERE side?

Hey @fviernau I found some usage of the old syntax by us. Let me fix them first please, I will then give you an heads-up.

**Comment by bennati** [comment]
please go ahead

**Comment by fviernau** [comment]
> Hey @fviernau I found some usage of the old syntax by us. Let me fix them first please, I will then give you an heads-up.

Thanks @nnobelis !

**Review by mnonnenmacher — APPROVED** [review]

**Comment by mnonnenmacher** [comment]
@fviernau Fixes on our side are done.

**Comment by MarcelBochtler** [comment]
Unfortunately, this breaks all projects, which use the legacy format in their `.ort.yml` :/

**Comment by MarcelBochtler** [comment]
Additionally, the JsonSchema for the curations still shows the legacy format as valid.

**Comment by mnonnenmacher** [comment]
I guess the problem is that the documentation was not updated long ago when the format was changed in 2019.

**Pull request merged**